fix(transformer/typescript): remove type-only `import =` when `only_remove_type_imports` is true

Dunqing opened the pull request:
close: https://github.com/oxc-project/oxc/issues/8230
close: https://github.com/rolldown/rolldown/issues/3287

Related PR in Babel: https://github.com/oxc-project/oxc/issues/8230

I have compared our output with TypeScript, and it is the same as `TypeScript`, Babel's implementation currently hasn't removed imports referenced by type-only `TSImportEqualsDeclaration`

Boshen commented on crates/oxc_transformer/src/typescript/module.rs:
```diff
@@ -7,12 +8,13 @@ use super::diagnostics;
 use crate::TransformCtx;
 
 pub struct TypeScriptModule<'a, 'ctx> {
+    only_remove_type_imports: bool,
```
can you reference https://babeljs.io/docs/babel-preset-typescript#onlyremovetypeimports

underfin commented on tasks/transform_conformance/tests/babel-plugin-transform-typescript/test/fixtures/preserve-import-=/output.js:
```diff
@@ -0,0 +1,3 @@
+import nsa from "./module-a";
+var Foo = nsa.bar;
```
The `Foo` is not removed, it is a typing.

Dunqing replied:
`only_remove_type_imports` is enabled, so it only removes type-only imports. This option is the same as [verbatimModuleSyntax] (https://www.typescriptlang.org/tsconfig/#verbatimModuleSyntax) in `TypeScript`.

Dunqing replied:
See https://www.typescriptlang.org/play/?noLib=true&skipLibCheck=true&target=99&jsx=1&verbatimModuleSyntax=true#code/JYWwDg9gTgLgBAOwM4EM4DMoRHARAOgHoQIATAVwBsBTAWhVwG4AoUSWOAMQgjgF5EqfACMUUFswDGEZPHQ8AXFx784ABkZA

underfin replied:
Ok. I get you. It shoulde be remove if `only_remove_type_imports` is enabled ,right?

Dunqing replied:
Yes!

underfin commented:
The type-only `TSImportEqualsDeclaration` always should be removed, look like it is unrelated with `only_remove_type_imports`.

Boshen commented:
@Dunqing feel free to merge.

Boshen commented:
### Merge activity

* **Jan 11, 3:31 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(transformer/typescript): remove type-only `import =` when `only_remove_type_imports` is true (#8275)

close: https://github.com/oxc-project/oxc/issues/8230
close: https://github.com/rolldown/rolldown/issues/3287

Related PR in Babel: https://github.com/oxc-project/oxc/issues/8230

I have compared our output with TypeScript, and it is the same as `TypeScript`, Babel's implementation currently hasn't removed imports referenced by type-only `TSImportEqualsDeclaration`
